exchanges/persisted-fetch/src/sha256.ts

```diff
@@ -31,7 +31,17 @@ const sha256Browser = (bytes: Uint8Array): Promise<Uint8Array> => {
 };
 
 const nodeCrypto =
-  typeof window === 'undefined' ? eval("require('crypto')") : null;
+  typeof window === 'undefined'
+    ? (() => {
+        try {
+          // Call eval indirectly to guarantee no side-effects in module scope
+          // (optimization for minifiers)
+          return (0, eval)("r => r('crypto')")(require);
```

Comment:
I originally wanted to put this here:
```js
(0, eval)("require('crypto')");
```
But, apparently global scope doesn't have `require`. It only exists in module scope, so passing a reference is needed.

Reply:
I think we'll want to test this with a multitude of bundlers and options today, namely Rollup and some other solutions on top.

While I'm pretty confident that Webpack won't mind this, I do think we'll want to play this one safely

Reply:
Ok, haven't quite gotten to testing this yet, but I'll aready note down that this will require transpilation, which may break in non obvious ways, so a simpler written form may be: 

```suggestion
return (new Function('require', 'return require("crypto")'))(require);
```

Reply:
I just tested this version out:
```typescript
let nodeCrypto;
if (typeof window === 'undefined') {
  try {
    // Indirect eval/require to guarantee no side-effects in module scope
    // (optimization for minifiers)
    nodeCrypto = new Function('require', 'return require("crypto")')(require);
  } catch (e) {}
}
```

And confirmed that it accomplishes the same thing. The function constructor doesn't inherit local scope.

This brings our sizes down to:

* `urql-exchange-persisted-fetch.min.js` -> `2783b`
* `urql-exchange-persisted-fetch.min.mjs` -> `2809b`

Small, but still a gain 💪 (and it looks nicer than the evaled-IIFE inside an IIFE inside a ternary)

I haven't tested it with `rollup` (which was the source of the original issue raised a few months ago), but this clears out a warning I was seeing in `esbuild`

Updated here: b13b3633